src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -872,10 +872,11 @@ private void Get_CORINFO_SIG_INFO(MethodSignature signature, CORINFO_SIG_INFO* s
             sig->sigInst.methInst = null; // Not used by the JIT
             sig->sigInst.methInstCount = (uint)signature.GenericParameterCount;
 
+            MethodILScope scope = _compilation.GetMethodIL(MethodBeingCompiled);
```

Comment:
MethodBeingCompiler will not be the right scope if we're inlining.

Reply:
Do you have any recommendations about how to get the right scope? I'm not sure how to get the right one here in all cases.

Reply:
Which callsite to `Get_CORINFO_SIG_INFO` looks problematic? Worst case we need a JitInterface change. Or we could also smuggle the "disable marshalling" bit in `sig->flags` if getting the scope is hard.

Reply:
I think we can thread through the scope from the caller. I'll try that out.